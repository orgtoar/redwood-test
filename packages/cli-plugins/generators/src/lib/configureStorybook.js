import path from 'path'
import util from 'util'

import fse from 'fs-extra'
import prettier from 'prettier'

import { merge } from './merge'
import {
  interleave,
  concatUnique,
  keepBoth,
  keepBothStatementParents,
} from './merge/strategy'

import { getPaths } from '.'

export default async function extendStorybookConfiguration(
  newConfigPath = undefined
) {
  const sbPreviewConfigPath = getPaths().web.storybookPreviewConfig
  if (!fse.existsSync(sbPreviewConfigPath)) {
    await util.promisify(fse.cp)(
      path.join(__dirname, 'templates', 'storybook.preview.js.template'),
      sbPreviewConfigPath
    )
  }

  if (newConfigPath) {
    const read = (path) => fse.readFileSync(path, { encoding: 'utf-8' })
    const write = (path, data) => fse.writeFileSync(path, data)
    const merged = merge(read(sbPreviewConfigPath), read(newConfigPath), {
      ImportDeclaration: interleave,
      ArrayExpression: concatUnique,
      ObjectExpression: concatUnique,
      ArrowFunctionExpression: keepBothStatementParents,
      FunctionDeclaration: keepBoth,
    })

    const formatted = prettier.format(merged, {
      parser: 'babel',
      ...(await prettier.resolveConfig(sbPreviewConfigPath)),
    })

    write(sbPreviewConfigPath, formatted)
  }
}
