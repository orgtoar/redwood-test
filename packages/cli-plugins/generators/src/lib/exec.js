import path from 'path'

import { registerApiSideBabelHook } from '@redwoodjs/internal/dist/build/babel/api'
import { getWebSideDefaultBabelConfig } from '@redwoodjs/internal/dist/build/babel/web'
import { getPaths } from '@redwoodjs/project-config'

export async function runScriptFunction({
  path: scriptPath,
  functionName,
  args,
}) {
  const script = await import(scriptPath)
  const returnValue = await script[functionName](args)

  try {
    const { db } = await import(path.join(getPaths().api.lib, 'db'))
    db.$disconnect()
  } catch (e) {
    // silence
  }

  return returnValue
}

export async function configureBabel() {
  const {
    overrides: _overrides,
    plugins: webPlugins,
    ...otherWebConfig
  } = getWebSideDefaultBabelConfig()

  // Import babel config for running script
  registerApiSideBabelHook({
    plugins: [
      [
        'babel-plugin-module-resolver',
        {
          alias: {
            $api: getPaths().api.base,
            $web: getPaths().web.base,
            api: getPaths().api.base,
            web: getPaths().web.base,
          },
          loglevel: 'silent', // to silence the unnecessary warnings
        },
        'exec-$side-module-resolver',
      ],
    ],
    overrides: [
      {
        test: ['./api/'],
        plugins: [
          [
            'babel-plugin-module-resolver',
            {
              alias: {
                src: getPaths().api.src,
              },
              loglevel: 'silent',
            },
            'exec-api-src-module-resolver',
          ],
        ],
      },
      {
        test: ['./web/'],
        plugins: [
          ...webPlugins,
          [
            'babel-plugin-module-resolver',
            {
              alias: {
                src: getPaths().web.src,
              },
              loglevel: 'silent',
            },
            'exec-web-src-module-resolver',
          ],
        ],
        ...otherWebConfig,
      },
    ],
  })
}
