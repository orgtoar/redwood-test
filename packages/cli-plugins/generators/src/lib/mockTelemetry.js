/* eslint-env jest */

// mock Telemetry for CLI commands so they don't try to spawn a process
jest.mock('@redwoodjs/telemetry', () => {
  return {
    errorTelemetry: () => jest.fn(),
    timedTelemetry: () => jest.fn(),
  }
})
