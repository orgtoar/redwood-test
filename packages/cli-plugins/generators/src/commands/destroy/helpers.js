import { Listr } from 'listr2'

import { deleteFilesTask } from '../../lib'
import c from '../../lib/colors'

const tasks = ({ componentName, filesFn, name }) =>
  new Listr(
    [
      {
        title: `Destroying ${componentName} files...`,
        task: async () => {
          const f = await filesFn({ name, stories: true, tests: true })
          return deleteFilesTask(f)
        },
      },
    ],
    { rendererOptions: { collapseSubtasks: false }, exitOnError: true }
  )

export const createYargsForComponentDestroy = ({
  componentName,
  preTasksFn = (options) => options,
  filesFn,
}) => {
  return {
    command: `${componentName} <name>`,
    description: `Destroy a ${componentName} component`,
    builder: (yargs) => {
      yargs.positional('name', {
        description: `Name of the ${componentName}`,
        type: 'string',
      })
    },
    handler: async (options) => {
      try {
        options = await preTasksFn({ ...options, isDestroyer: true })
        await tasks({ componentName, filesFn, name: options.name }).run()
      } catch (e) {
        console.log(c.error(e.message))
      }
    },
    tasks,
  }
}
