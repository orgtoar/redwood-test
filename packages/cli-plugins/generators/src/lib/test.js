/* eslint-env jest */

// Include at the top of your tests. Automatically mocks out the file system
//
// import { loadComponentFixture } from 'src/lib/test'
//
// test('true is true', () => {
//   expect('some output').toEqual(loadComponentFixture('component', 'filename.js'))
// })

import fs from 'fs'
import path from 'path'

import './mockTelemetry'

jest.mock('@redwoodjs/internal/dist/generate/generate', () => {
  return {
    generate: () => {},
  }
})

jest.mock('@redwoodjs/project-config', () => {
  const path = require('path')
  return {
    ...jest.requireActual('@redwoodjs/project-config'),
    getPaths: () => {
      const BASE_PATH = '/path/to/project'
      return {
        base: BASE_PATH,
        api: {
          dataMigrations: path.join(BASE_PATH, './api/prisma/dataMigrations'),
          db: path.join(globalThis.__dirname, 'fixtures'), // this folder
          dbSchema: path.join(
            globalThis.__dirname,
            'fixtures',
            'schema.prisma'
          ), // this folder
          generators: path.join(BASE_PATH, './api/generators'),
          src: path.join(BASE_PATH, './api/src'),
          services: path.join(BASE_PATH, './api/src/services'),
          directives: path.join(BASE_PATH, './api/src/directives'),
          graphql: path.join(BASE_PATH, './api/src/graphql'),
          functions: path.join(BASE_PATH, './api/src/functions'),
        },
        web: {
          config: path.join(BASE_PATH, './web/config'),
          src: path.join(BASE_PATH, './web/src'),
          generators: path.join(BASE_PATH, './web/generators'),
          routes: path.join(BASE_PATH, 'web/src/Routes.js'),
          components: path.join(BASE_PATH, '/web/src/components'),
          layouts: path.join(BASE_PATH, '/web/src/layouts'),
          pages: path.join(BASE_PATH, '/web/src/pages'),
          app: path.join(BASE_PATH, '/web/src/App.js'),
        },
        scripts: path.join(BASE_PATH, 'scripts'),
        generated: {
          base: path.join(BASE_PATH, '.redwood'),
          schema: path.join(BASE_PATH, '.redwood/schema.graphql'),
          types: {
            includes: path.join(BASE_PATH, '.redwood/types/includes'),
            mirror: path.join(BASE_PATH, '.redwood/types/mirror'),
          },
        },
      }
    },
  }
})

jest.mock('./project', () => ({
  isTypeScriptProject: () => false,
  sides: () => ['web', 'api'],
}))

globalThis.__prettierPath = path.resolve(
  __dirname,
  './__tests__/fixtures/prettier.config.js'
)

jest.mock('path', () => {
  const path = jest.requireActual('path')
  return {
    ...path,
    join: (...paths) => {
      if (
        paths &&
        paths[0] === '/path/to/project' &&
        paths[1] === 'prettier.config.js'
      ) {
        return globalThis.__prettierPath
      }
      return path.join(...paths)
    },
  }
})

jest.spyOn(Math, 'random').mockReturnValue(0.123456789)

export const generatorsRootPath = path.join(
  __dirname,
  '..',
  'commands',
  'generate'
)

/**
 * Loads the fixture for a generator by assuming a lot of the path structure
 * automatically:
 *
 *   `loadGeneratorFixture('scaffold', 'NamePage.js')`
 *
 * will return the contents of:
 *
 *   `cli/src/commands/generate/scaffold/__tests__/fixtures/NamePage.js`
 */
export const loadGeneratorFixture = (generator, name) => {
  return loadFixture(
    path.join(
      __dirname,
      '..',
      'commands',
      'generate',
      generator,
      '__tests__',
      'fixtures',
      name
    )
  )
}

/**
 * Returns the contents of a text file in a `fixtures` directory
 */
export const loadFixture = (filepath) => {
  return fs.readFileSync(filepath).toString()
}
