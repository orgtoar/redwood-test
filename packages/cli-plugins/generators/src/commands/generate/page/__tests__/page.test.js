globalThis.__dirname = __dirname

globalThis.mockFs = false
let mockFiles = {}

jest.mock('fs', () => {
  const actual = jest.requireActual('fs')

  return {
    ...actual,
    existsSync: (...args) => {
      if (!globalThis.mockFs) {
        return actual.existsSync.apply(null, args)
      }
      return false
    },
    mkdirSync: (...args) => {
      if (!globalThis.mockFs) {
        return actual.mkdirSync.apply(null, args)
      }
    },
    writeFileSync: (target, contents) => {
      if (!globalThis.mockFs) {
        return actual.writeFileSync.call(null, target, contents)
      }
    },
    readFileSync: (path) => {
      if (!globalThis.mockFs) {
        return actual.readFileSync.call(null, path)
      }

      const mockedContent = mockFiles[path]

      return mockedContent || actual.readFileSync.call(null, path)
    },
  }
})

import fs from 'fs'
import path from 'path'

// Load mocks
import '../../../../lib/test'

import { ensurePosixPath } from '@redwoodjs/project-config'

import { getPaths } from '../../../../lib'
import { pathName } from '../../helpers'
import * as page from '../page'

describe('Single world files', () => {
  const singleWordFiles = page.files({
    name: 'Home',
    tests: true,
    stories: true,
    ...page.paramVariants(pathName(undefined, 'home')),
  })

  it('returns exactly 3 files', () => {
    expect(Object.keys(singleWordFiles).length).toEqual(3)
  })

  it('creates a page component', () => {
    expect(
      singleWordFiles[
        path.normalize('/path/to/project/web/src/pages/HomePage/HomePage.js')
      ]
    ).toMatchSnapshot()
  })

  it('creates a page test', () => {
    expect(
      singleWordFiles[
        path.normalize(
          '/path/to/project/web/src/pages/HomePage/HomePage.test.js'
        )
      ]
    ).toMatchSnapshot()
  })

  it('creates a page story', () => {
    expect(
      singleWordFiles[
        path.normalize(
          '/path/to/project/web/src/pages/HomePage/HomePage.stories.js'
        )
      ]
    ).toMatchSnapshot()
  })
})

describe('multiWorldFiles', () => {
  const multiWordFiles = page.files({
    name: 'ContactUs',
    tests: true,
    stories: true,
    ...page.paramVariants(pathName(undefined, 'contact-us')),
  })

  it('creates a page component', () => {
    expect(
      multiWordFiles[
        path.normalize(
          '/path/to/project/web/src/pages/ContactUsPage/ContactUsPage.js'
        )
      ]
    ).toMatchSnapshot()
  })

  it('creates a test for a component with multiple words for a name', () => {
    expect(
      multiWordFiles[
        path.normalize(
          '/path/to/project/web/src/pages/ContactUsPage/ContactUsPage.test.js'
        )
      ]
    ).toMatchSnapshot()
  })

  it('creates a page story', () => {
    expect(
      multiWordFiles[
        path.normalize(
          '/path/to/project/web/src/pages/ContactUsPage/ContactUsPage.stories.js'
        )
      ]
    ).toMatchSnapshot()
  })
})

describe('Plural word files', () => {
  const pluralWordFiles = page.files({
    name: 'Cats',
    tests: true,
    stories: true,
    ...page.paramVariants(pathName(undefined, 'cats')),
  })

  test('creates a page component with a plural word for name', () => {
    expect(
      pluralWordFiles[
        path.normalize('/path/to/project/web/src/pages/CatsPage/CatsPage.js')
      ]
    ).toMatchSnapshot()
  })
})

describe('paramFiles', () => {
  const paramFiles = page.files({
    name: 'Post',
    tests: true,
    stories: true,
    ...page.paramVariants(pathName('{id}', 'post')),
  })

  it('creates a page component with params', () => {
    expect(
      paramFiles[
        path.normalize('/path/to/project/web/src/pages/PostPage/PostPage.js')
      ]
    ).toMatchSnapshot()
  })

  it('creates a test for page component with params', () => {
    expect(
      paramFiles[
        path.normalize(
          '/path/to/project/web/src/pages/PostPage/PostPage.test.js'
        )
      ]
    ).toMatchSnapshot()
  })
})

describe('No test files', () => {
  const noTestsFiles = page.files({
    name: 'NoTests',
    tests: false,
    stories: true,
    ...page.paramVariants(pathName(undefined, 'no-tests')),
  })

  it('doesnt create a test for page component when tests=false', () => {
    expect(Object.keys(noTestsFiles)).toEqual([
      path.normalize(
        '/path/to/project/web/src/pages/NoTestsPage/NoTestsPage.stories.js'
      ),
      path.normalize(
        '/path/to/project/web/src/pages/NoTestsPage/NoTestsPage.js'
      ),
    ])
  })
})

describe('No stories files', () => {
  const noStoriesFiles = page.files({
    name: 'NoStories',
    tests: true,
    stories: false,
    ...page.paramVariants(pathName(undefined, 'no-stories')),
  })

  it('doesnt create a story for page component when stories=false', () => {
    expect(Object.keys(noStoriesFiles)).toEqual([
      path.normalize(
        '/path/to/project/web/src/pages/NoStoriesPage/NoStoriesPage.test.js'
      ),
      path.normalize(
        '/path/to/project/web/src/pages/NoStoriesPage/NoStoriesPage.js'
      ),
    ])
  })
})

test('creates a single-word route name', () => {
  const names = ['Home', 'home']

  names.forEach((name) => {
    expect(page.routes({ name: name, path: '/' })).toEqual([
      '<Route path="/" page={HomePage} name="home" />',
    ])
  })
})

test('creates a camelCase route name for lowercase words', () => {
  const names = ['FooBar', 'foo_bar', 'foo-bar', 'fooBar']

  names.forEach((name) => {
    expect(page.routes({ name: name, path: 'foo-bar' })).toEqual([
      '<Route path="foo-bar" page={FooBarPage} name="fooBar" />',
    ])
  })
})

test('creates a camelCase route name for uppercase words', () => {
  const names = ['FOO_BAR', 'FOO-BAR']

  names.forEach((name) => {
    expect(page.routes({ name: name, path: 'foo-bar' })).toEqual([
      '<Route path="foo-bar" page={FOOBARPage} name="fooBar" />',
    ])
  })
})

test('creates a camelCase route name for uppercase and lowercase mixed words', () => {
  const names = ['FOOBar', 'FOO-Bar', 'FOO_Bar']

  names.forEach((name) => {
    expect(page.routes({ name: name, path: 'foo-bar' })).toEqual([
      '<Route path="foo-bar" page={FOOBarPage} name="fooBar" />',
    ])
  })
})

test('creates a camelCase route name for multiple word names', () => {
  const names = ['AbTest', 'abTest', 'ab-test', 'ab_test']

  names.forEach((name) => {
    expect(page.routes({ name: name, path: 'foo-bar' })).toEqual([
      '<Route path="foo-bar" page={AbTestPage} name="abTest" />',
    ])
  })
})

test('creates a camelCase route name for multiple words with uppercase character after special character', () => {
  const names = ['ABtest', 'aBtest', 'a-Btest', 'a_Btest']

  names.forEach((name) => {
    expect(page.routes({ name: name, path: 'foo-bar' })).toEqual([
      '<Route path="foo-bar" page={ABtestPage} name="aBtest" />',
    ])
  })
})

test('creates a camelCase route name for multiple words starting with uppercase characters', () => {
  const names = ['ABTest', 'AB_test', 'AB-test']

  names.forEach((name) => {
    expect(page.routes({ name: name, path: 'foo-bar' })).toEqual([
      '<Route path="foo-bar" page={ABTestPage} name="abTest" />',
    ])
  })
})

test('creates a path equal to passed path', () => {
  expect(page.routes({ name: 'FooBar', path: 'fooBar-baz' })).toEqual([
    '<Route path="fooBar-baz" page={FooBarPage} name="fooBar" />',
  ])
})

test('paramVariants returns empty strings for no params', () => {
  const emptyParams = {
    propParam: '',
    propValueParam: '',
    argumentParam: '',
    paramName: '',
    paramValue: '',
    paramType: '',
  }
  expect(page.paramVariants()).toEqual(emptyParams)
  expect(page.paramVariants('')).toEqual(emptyParams)
  expect(page.paramVariants('/')).toEqual(emptyParams)
  expect(page.paramVariants('/post/edit')).toEqual(emptyParams)
})

test('paramVariants finds the param and type in the middle of the path', () => {
  expect(page.paramVariants('/post/{id:Int}/edit')).toEqual({
    propParam: '{ id }',
    propValueParam: 'id={42}',
    argumentParam: '{ id: 42 }',
    paramName: 'id',
    paramValue: 42,
    paramType: 'number',
  })
})

test('paramVariants handles boolean type', () => {
  expect(page.paramVariants('/post/edit/{debug:Boolean}')).toEqual({
    propParam: '{ debug }',
    propValueParam: 'debug={true}',
    argumentParam: '{ debug: true }',
    paramName: 'debug',
    paramValue: true,
    paramType: 'boolean',
  })
})

test('paramVariants paramType defaults to string', () => {
  expect(page.paramVariants('/posts/{id}')).toEqual({
    propParam: '{ id }',
    propValueParam: "id={'42'}",
    argumentParam: "{ id: '42' }",
    paramName: 'id',
    paramValue: '42',
    paramType: 'string',
  })
})

describe('handler', () => {
  beforeEach(() => {
    jest.spyOn(console, 'info').mockImplementation(() => {})
    jest.spyOn(console, 'log').mockImplementation(() => {})
  })

  afterEach(() => {
    console.info.mockRestore()
    console.log.mockRestore()
  })

  test('file generation', async () => {
    mockFiles = {
      [getPaths().web.routes]: [
        "import { Router, Route } from '@redwoodjs/router'",
        '',
        'const Routes = () => {',
        '  return (',
        '    <Router>',
        '      <Route path="/about" page={AboutPage} name="about" />',
        '      <Route notfound page={NotFoundPage} />',
        '    </Router>',
        '  )',
        '}',
        '',
        'export default Routes',
      ].join('\n'),
    }

    const spy = jest.spyOn(fs, 'writeFileSync')

    globalThis.mockFs = true

    await page.handler({
      name: 'HomePage', // 'Page' should be trimmed from name
      path: '',
      force: false,
      tests: true,
      stories: true,
    })

    expect(spy).toHaveBeenCalled()

    spy.mock.calls.forEach((calls) => {
      const testOutput = {
        // Because windows paths are different, we need to normalise before snapshotting
        filePath: ensurePosixPath(calls[0]),
        fileContent: calls[1],
      }
      expect(testOutput).toMatchSnapshot()
    })

    globalThis.mockFs = false
    spy.mockRestore()
  })

  test('file generation with route params', async () => {
    mockFiles = {
      [getPaths().web.routes]: [
        "import { Router, Route } from '@redwoodjs/router'",
        '',
        'const Routes = () => {',
        '  return (',
        '    <Router>',
        '      <Route path="/about" page={AboutPage} name="about" />',
        '      <Route notfound page={NotFoundPage} />',
        '    </Router>',
        '  )',
        '}',
        '',
        'export default Routes',
      ].join('\n'),
    }

    const spy = jest.spyOn(fs, 'writeFileSync')
    globalThis.mockFs = true

    await page.handler({
      name: 'post',
      path: '{id}',
      force: false,
      tests: true,
      stories: true,
    })

    expect(spy).toHaveBeenCalled()

    spy.mock.calls.forEach((calls) => {
      const testOutput = {
        filePath: ensurePosixPath(calls[0]),
        fileContent: calls[1],
      }
      expect(testOutput).toMatchSnapshot()
    })

    globalThis.mockFs = false
    spy.mockRestore()
  })
})

describe('TS Files', () => {
  const typescriptFiles = page.files({
    name: 'TSFiles',
    typescript: true,
    tests: true,
    stories: true,
    ...page.paramVariants(pathName(undefined, 'typescript')),
  }) //?

  it('generates typescript pages', () => {
    expect(
      typescriptFiles[
        path.normalize(
          '/path/to/project/web/src/pages/TSFilesPage/TSFilesPage.tsx'
        )
      ]
    ).toMatchSnapshot()

    expect(
      typescriptFiles[
        path.normalize(
          '/path/to/project/web/src/pages/TSFilesPage/TSFilesPage.stories.tsx'
        )
      ]
    ).toMatchSnapshot()

    expect(
      typescriptFiles[
        path.normalize(
          '/path/to/project/web/src/pages/TSFilesPage/TSFilesPage.test.tsx'
        )
      ]
    ).toMatchSnapshot()
  })

  test('TS Params', () => {
    const typescriptParamFiles = page.files({
      name: 'TSParamFiles',
      typescript: true,
      tests: true,
      stories: true,
      ...page.paramVariants(pathName('{id}', 'typescript-param')),
    })

    expect(
      typescriptParamFiles[
        path.normalize(
          '/path/to/project/web/src/pages/TSParamFilesPage/TSParamFilesPage.tsx'
        )
      ]
    ).toMatchSnapshot()
  })

  test('TS Params with type', () => {
    const typescriptParamTypeFiles = page.files({
      name: 'TSParamTypeFiles',
      typescript: true,
      tests: false,
      stories: false,
      ...page.paramVariants(
        pathName('/bazinga-ts/{id:Int}', 'typescript-param-with-type')
      ),
    })

    expect(
      typescriptParamTypeFiles[
        path.normalize(
          '/path/to/project/web/src/pages/TSParamTypeFilesPage/TSParamTypeFilesPage.tsx'
        )
      ]
    ).toMatchSnapshot()
  })
})
