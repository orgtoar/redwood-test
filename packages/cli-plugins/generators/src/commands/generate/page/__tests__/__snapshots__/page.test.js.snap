// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Plural word files creates a page component with a plural word for name 1`] = `
"import { Link, routes } from '@redwoodjs/router'
import { MetaTags } from '@redwoodjs/web'

const CatsPage = () => {
  return (
    <>
      <MetaTags title="Cats" description="Cats page" />

      <h1>CatsPage</h1>
      <p>
        Find me in <code>./web/src/pages/CatsPage/CatsPage.js</code>
      </p>
      <p>
        My default route is named <code>cats</code>, link to me with \`
        <Link to={routes.cats()}>Cats</Link>\`
      </p>
    </>
  )
}

export default CatsPage
"
`;

exports[`Single world files creates a page component 1`] = `
"import { Link, routes } from '@redwoodjs/router'
import { MetaTags } from '@redwoodjs/web'

const HomePage = () => {
  return (
    <>
      <MetaTags title="Home" description="Home page" />

      <h1>HomePage</h1>
      <p>
        Find me in <code>./web/src/pages/HomePage/HomePage.js</code>
      </p>
      <p>
        My default route is named <code>home</code>, link to me with \`
        <Link to={routes.home()}>Home</Link>\`
      </p>
    </>
  )
}

export default HomePage
"
`;

exports[`Single world files creates a page story 1`] = `
"import HomePage from './HomePage'

export const generated = () => {
  return <HomePage />
}

export default {
  title: 'Pages/HomePage',
  component: HomePage,
}
"
`;

exports[`Single world files creates a page test 1`] = `
"import { render } from '@redwoodjs/testing/web'

import HomePage from './HomePage'

//   Improve this test with help from the Redwood Testing Doc:
//   https://redwoodjs.com/docs/testing#testing-pages-layouts

describe('HomePage', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<HomePage />)
    }).not.toThrow()
  })
})
"
`;

exports[`TS Files TS Params 1`] = `
"import { Link, routes } from '@redwoodjs/router'
import { MetaTags } from '@redwoodjs/web'

type TsParamFilesPageProps = {
  id: string
}

const TsParamFilesPage = ({ id }: TsParamFilesPageProps) => {
  return (
    <>
      <MetaTags title="TsParamFiles" description="TsParamFiles page" />

      <h1>TsParamFilesPage</h1>
      <p>
        Find me in{' '}
        <code>./web/src/pages/TSParamFilesPage/TSParamFilesPage.tsx</code>
      </p>
      <p>
        My default route is named <code>tsParamFiles</code>, link to me with \`
        <Link to={routes.tsParamFiles({ id: '42' })}>TsParamFiles 42</Link>\`
      </p>
      <p>The parameter passed to me is {id}</p>
    </>
  )
}

export default TsParamFilesPage
"
`;

exports[`TS Files TS Params with type 1`] = `
"import { Link, routes } from '@redwoodjs/router'
import { MetaTags } from '@redwoodjs/web'

type TsParamTypeFilesPageProps = {
  id: number
}

const TsParamTypeFilesPage = ({ id }: TsParamTypeFilesPageProps) => {
  return (
    <>
      <MetaTags title="TsParamTypeFiles" description="TsParamTypeFiles page" />

      <h1>TsParamTypeFilesPage</h1>
      <p>
        Find me in{' '}
        <code>
          ./web/src/pages/TSParamTypeFilesPage/TSParamTypeFilesPage.tsx
        </code>
      </p>
      <p>
        My default route is named <code>tsParamTypeFiles</code>, link to me with
        \`
        <Link to={routes.tsParamTypeFiles({ id: 42 })}>
          TsParamTypeFiles 42
        </Link>
        \`
      </p>
      <p>The parameter passed to me is {id}</p>
    </>
  )
}

export default TsParamTypeFilesPage
"
`;

exports[`TS Files generates typescript pages 1`] = `
"import { Link, routes } from '@redwoodjs/router'
import { MetaTags } from '@redwoodjs/web'

const TsFilesPage = () => {
  return (
    <>
      <MetaTags title="TsFiles" description="TsFiles page" />

      <h1>TsFilesPage</h1>
      <p>
        Find me in <code>./web/src/pages/TSFilesPage/TSFilesPage.tsx</code>
      </p>
      <p>
        My default route is named <code>tsFiles</code>, link to me with \`
        <Link to={routes.tsFiles()}>TsFiles</Link>\`
      </p>
    </>
  )
}

export default TsFilesPage
"
`;

exports[`TS Files generates typescript pages 2`] = `
"import type { ComponentMeta } from '@storybook/react'

import TsFilesPage from './TsFilesPage'

export const generated = () => {
  return <TsFilesPage />
}

export default {
  title: 'Pages/TsFilesPage',
  component: TsFilesPage,
} as ComponentMeta<typeof TsFilesPage>
"
`;

exports[`TS Files generates typescript pages 3`] = `
"import { render } from '@redwoodjs/testing/web'

import TsFilesPage from './TsFilesPage'

//   Improve this test with help from the Redwood Testing Doc:
//   https://redwoodjs.com/docs/testing#testing-pages-layouts

describe('TsFilesPage', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<TsFilesPage />)
    }).not.toThrow()
  })
})
"
`;

exports[`handler file generation 1`] = `
{
  "fileContent": "import HomePage from './HomePage'

export const generated = () => {
  return <HomePage />
}

export default {
  title: 'Pages/HomePage',
  component: HomePage,
}
",
  "filePath": "/path/to/project/web/src/pages/HomePage/HomePage.stories.js",
}
`;

exports[`handler file generation 2`] = `
{
  "fileContent": "import { render } from '@redwoodjs/testing/web'

import HomePage from './HomePage'

//   Improve this test with help from the Redwood Testing Doc:
//   https://redwoodjs.com/docs/testing#testing-pages-layouts

describe('HomePage', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<HomePage />)
    }).not.toThrow()
  })
})
",
  "filePath": "/path/to/project/web/src/pages/HomePage/HomePage.test.js",
}
`;

exports[`handler file generation 3`] = `
{
  "fileContent": "import { Link, routes } from '@redwoodjs/router'
import { MetaTags } from '@redwoodjs/web'

const HomePage = () => {
  return (
    <>
      <MetaTags title="Home" description="Home page" />

      <h1>HomePage</h1>
      <p>
        Find me in <code>./web/src/pages/HomePage/HomePage.js</code>
      </p>
      <p>
        My default route is named <code>home</code>, link to me with \`
        <Link to={routes.home()}>Home</Link>\`
      </p>
    </>
  )
}

export default HomePage
",
  "filePath": "/path/to/project/web/src/pages/HomePage/HomePage.js",
}
`;

exports[`handler file generation 4`] = `
{
  "fileContent": "import { Router, Route } from '@redwoodjs/router'

const Routes = () => {
  return (
    <Router>
      <Route path="/home" page={HomePage} name="home" />
      <Route path="/about" page={AboutPage} name="about" />
      <Route notfound page={NotFoundPage} />
    </Router>
  )
}

export default Routes",
  "filePath": "/path/to/project/web/src/Routes.js",
}
`;

exports[`handler file generation with route params 1`] = `
{
  "fileContent": "import PostPage from './PostPage'

export const generated = (args) => {
  return <PostPage id={'42'} {...args} />
}

export default {
  title: 'Pages/PostPage',
  component: PostPage,
}
",
  "filePath": "/path/to/project/web/src/pages/PostPage/PostPage.stories.js",
}
`;

exports[`handler file generation with route params 2`] = `
{
  "fileContent": "import { render } from '@redwoodjs/testing/web'

import PostPage from './PostPage'

//   Improve this test with help from the Redwood Testing Doc:
//   https://redwoodjs.com/docs/testing#testing-pages-layouts

describe('PostPage', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<PostPage id={'42'} />)
    }).not.toThrow()
  })
})
",
  "filePath": "/path/to/project/web/src/pages/PostPage/PostPage.test.js",
}
`;

exports[`handler file generation with route params 3`] = `
{
  "fileContent": "import { Link, routes } from '@redwoodjs/router'
import { MetaTags } from '@redwoodjs/web'

const PostPage = ({ id }) => {
  return (
    <>
      <MetaTags title="Post" description="Post page" />

      <h1>PostPage</h1>
      <p>
        Find me in <code>./web/src/pages/PostPage/PostPage.js</code>
      </p>
      <p>
        My default route is named <code>post</code>, link to me with \`
        <Link to={routes.post({ id: '42' })}>Post 42</Link>\`
      </p>
      <p>The parameter passed to me is {id}</p>
    </>
  )
}

export default PostPage
",
  "filePath": "/path/to/project/web/src/pages/PostPage/PostPage.js",
}
`;

exports[`handler file generation with route params 4`] = `
{
  "fileContent": "import { Router, Route } from '@redwoodjs/router'

const Routes = () => {
  return (
    <Router>
      <Route path="/post/{id}" page={PostPage} name="post" />
      <Route path="/about" page={AboutPage} name="about" />
      <Route notfound page={NotFoundPage} />
    </Router>
  )
}

export default Routes",
  "filePath": "/path/to/project/web/src/Routes.js",
}
`;

exports[`multiWorldFiles creates a page component 1`] = `
"import { Link, routes } from '@redwoodjs/router'
import { MetaTags } from '@redwoodjs/web'

const ContactUsPage = () => {
  return (
    <>
      <MetaTags title="ContactUs" description="ContactUs page" />

      <h1>ContactUsPage</h1>
      <p>
        Find me in <code>./web/src/pages/ContactUsPage/ContactUsPage.js</code>
      </p>
      <p>
        My default route is named <code>contactUs</code>, link to me with \`
        <Link to={routes.contactUs()}>ContactUs</Link>\`
      </p>
    </>
  )
}

export default ContactUsPage
"
`;

exports[`multiWorldFiles creates a page story 1`] = `
"import ContactUsPage from './ContactUsPage'

export const generated = () => {
  return <ContactUsPage />
}

export default {
  title: 'Pages/ContactUsPage',
  component: ContactUsPage,
}
"
`;

exports[`multiWorldFiles creates a test for a component with multiple words for a name 1`] = `
"import { render } from '@redwoodjs/testing/web'

import ContactUsPage from './ContactUsPage'

//   Improve this test with help from the Redwood Testing Doc:
//   https://redwoodjs.com/docs/testing#testing-pages-layouts

describe('ContactUsPage', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<ContactUsPage />)
    }).not.toThrow()
  })
})
"
`;

exports[`paramFiles creates a page component with params 1`] = `
"import { Link, routes } from '@redwoodjs/router'
import { MetaTags } from '@redwoodjs/web'

const PostPage = ({ id }) => {
  return (
    <>
      <MetaTags title="Post" description="Post page" />

      <h1>PostPage</h1>
      <p>
        Find me in <code>./web/src/pages/PostPage/PostPage.js</code>
      </p>
      <p>
        My default route is named <code>post</code>, link to me with \`
        <Link to={routes.post({ id: '42' })}>Post 42</Link>\`
      </p>
      <p>The parameter passed to me is {id}</p>
    </>
  )
}

export default PostPage
"
`;

exports[`paramFiles creates a test for page component with params 1`] = `
"import { render } from '@redwoodjs/testing/web'

import PostPage from './PostPage'

//   Improve this test with help from the Redwood Testing Doc:
//   https://redwoodjs.com/docs/testing#testing-pages-layouts

describe('PostPage', () => {
  it('renders successfully', () => {
    expect(() => {
      render(<PostPage id={'42'} />)
    }).not.toThrow()
  })
})
"
`;
